NES/PPU.cpp

```diff
@@ -223,40 +223,32 @@ inline void PPU::emitPixel() {
             sprite.shift();
         }
     }
-    
+
     //When bg rendering is off
     if ((ppumask & 8) == 0) {
         paletteIndex = 0;
     }
-    
-    if (showSprite) {
-        u8 p = ppuread(0x3F00 | spritePaletteIndex) * 3;
-        u8 r = palette[p];
-        u8 g = palette[p + 1];
-        u8 b = palette[p + 2];
-        buffer[pixelIndex++] = 255 << 24 | r << 16 | g << 8 | b;
-    } else {
-        u8 p = ppuread(0x3F00 | paletteIndex) * 3;
-        u8 r = palette[p];
-        u8 g = palette[p + 1];
-        u8 b = palette[p + 2];
-        buffer[pixelIndex++] = 255 << 24 | r << 16 | g << 8 | b;
-    }
+
+    u8 p = ppuread(0x3F00 | showSprite ? spritePaletteIndex : paletteIndex) * 3;
```

Comment:
Nice! I must have been very unfocused when writing this! Thanks for fixing!

Reply:
A bracket was missing here around `showSprite ? spritePaletteIndex : paletteIndex` so the colors were messed up, fixed it.